app/code/Magento/Catalog/Model/Indexer/Product/Price/System/Config/PriceScope.php

```diff
@@ -31,12 +39,14 @@ public function __construct(
         \Magento\Framework\App\Config\ScopeConfigInterface $config,
         \Magento\Framework\App\Cache\TypeListInterface $cacheTypeList,
         \Magento\Framework\Indexer\IndexerRegistry $indexerRegistry,
+        \Magento\Catalog\Model\Config\PriceScopeChange $priceScopeChange,
```

Comment:
Please move it to the last argument  and make it optional with fallback to retrieving it from ObjectManager for backward compatibility reasons

Reply:
Updated the code as per the suggestion.